File: src/project_manager_flutter/lib/features/user/repository/user_repository.dart
import 'package:dio/dio.dart';
import '../../../core/constants/api_constants.dart';
import '../../../core/network/api_exceptions.dart';
import '../models/update_user_request.dart';
import '../models/user_profile.dart';

class UserRepository {
  final Dio _dio;

  UserRepository(this._dio);

  Future<UserProfile> getCurrentUser() async {
    try {
      final response = await _dio.get(ApiConstants.currentUser);
      return UserProfile.fromJson(response.data);
    } on DioException catch (e) {
      throw ApiException.fromResponse(
        e.response?.statusCode ?? 500,
        e.response?.data,
      );
    }
  }

  Future<UserProfile> updateUser(UpdateUserRequest request) async {
    try {
      final response = await _dio.put(
        ApiConstants.currentUser,
        data: request.toJson(),
      );
      return UserProfile.fromJson(response.data);
    } on DioException catch (e) {
      throw ApiException.fromResponse(
        e.response?.statusCode ?? 500,
        e.response?.data,
      );
    }
  }
}

File: src/project-manager-web/src/features/user/repository/user-repository.ts
import { apiClient } from "@/core/api/client";
import { ApiEndpoints } from "@/core/api/constants";
import { ApiException } from "@/core/api/exceptions";
import type { UserProfile, UpdateUserRequest } from "../models/types";
import { AxiosError } from "axios";

export const userRepository = {
  async getCurrentUser(): Promise<UserProfile> {
    try {
      const response = await apiClient.get<UserProfile>(ApiEndpoints.currentUser);
      return response.data;
    } catch (error) {
      if (error instanceof AxiosError) {
        throw ApiException.fromResponse(error.response?.status ?? 500, error.response?.data);
      }
      throw error;
    }
  },

  async updateUser(request: UpdateUserRequest): Promise<void> {
    try {
      await apiClient.put(ApiEndpoints.currentUser, request);
    } catch (error) {
      if (error instanceof AxiosError) {
        throw ApiException.fromResponse(error.response?.status ?? 500, error.response?.data);
      }
      throw error;
    }
  },
};
